fuel-vm/src/interpreter/blockchain.rs

```diff
@@ -420,26 +737,46 @@ where
 
         // validations passed, perform the mutations
 
-        self.base_asset_balance_sub(amount)?;
+        base_asset_balance_sub(self.balances, self.memory, self.amount_coins_to_send)?;
```

Comment:
this should subtract from a contract balance if the context is internal. Although this should be done in a separate PR.